lib/ansible/modules/cloud/amazon/ec2_vpc_nacl.py

```diff
@@ -297,13 +293,20 @@ def rules_changed(aws_rules, param_rules, Egress, nacl_id, client, module):
     return changed
 
 
-def process_rule_entry(entry, Egress):
+def process_rule_entry(entry, Egress, module):
     params = dict()
     params['RuleNumber'] = entry[0]
     params['Protocol'] = str(PROTOCOL_NUMBERS[entry[1]])
     params['RuleAction'] = entry[2]
     params['Egress'] = Egress
-    params['CidrBlock'] = entry[3]
+    match = CIDR_REGEX.match(entry[3])
+    if not match:
+        module.fail_json(msg="Invalid CIDR value: %s" % entry[3])
+    try:
+        socket.inet_pton(socket.AF_INET6, match.group(1))
+        params['Ipv6CidrBlock'] = entry[3]
+    except socket.error:
+        params['CidrBlock'] = entry[3]
```

Comment:
You could use the built in ip address verification from `module_utils` instead of using socket. You could also eliminate the `CIDR_REGEX` if you used the built in. i.e.

```python
from ansible.module_utils.network.common.utils import validate_ip_address
from ansible.module_utils.network.common.utils import validate_ip_v6_address
```
```python
if validate_ip_address(entry[3]):
    params['CidrBlock'] = entry[3]
elif validate_ip_v6_address(entry[3]):
    params['Ipv6CidrBlock'] = entry[3]
else:
    module.fail_json(msg="Invalid CIDR value: %s" % entry[3])
```

NOTE: `validate_ip_v6_address` is only available in Ansible 2.6 and up. So if this change is going to be backported you should account for that in the imports. These use these functions in this module for reference. https://github.com/ansible/ansible/blob/devel/lib/ansible/module_utils/network/f5/ipaddress.py

Reply:
Actually my mistake you would still need the CIDR_REGEX before passing to that validate function.

Reply:
We would need a validate_ip_v6_cidr() helper function, because validate_ip_v6_address does not validate CIDR notation. The utils module does not have such a function.

validate_ip_v6_address("fe80::20c:29ff:fe3c:55f1") returns True
validate_ip_v6_address("fe80::20c:29ff:fe3c:55f1/23") returns False

Reply:
yes a validate cidr helper function makes sense and then just checking the cidr portion of the `CIDR_REGEX` match is 0 to 32 or 0 to 128.

Reply:
It seems like it would have to come as a separate PR, being a shared module? Or do you think I can submit a change for ansible/module_utils? I'm concerned this may stall the PR.

Reply:
Oh yeah, what I meant was you could add the helper function in this module. If you or someone else then wants to move that code to a shared place in module_utils that would be a separate PR.

For now I think its worth using the two shared validate_ip_* functions in this module and wrapping a little code around those (in this module) to validate the CIDR as well. Thoughts?

Reply:
After searching a bit, I see interesting modules such as lib/ansible/module_utils/compat/ipaddress.py

Reply:
I put a helper function. It turns out the ipaddress module rejects values such as '0.0.0.0/0' and it seems to have a bug for the number of digits in the IPv6 cases.